File: src/index/executables/checkDataset.cpp
#include <cstdlib>
#include <iostream>
#include "chm_tools/Dataset.hpp"
namespace fs = chm::fs;

#ifndef SRC_DIR
	constexpr auto SRC_DIR = "";
#endif

int main() {
	try {
		const auto dataDir = fs::absolute(fs::path(SRC_DIR)) / "data";
		const auto datasetPath = dataDir / "test.bin";
		const auto descPath = dataDir / "testCpp.txt";
		chm::Dataset(datasetPath).writeLongDescription(descPath);
		std::cout << "Description of dataset " << datasetPath << " written to " << descPath << ".\n";

	} catch(const std::exception& e) {
		std::cerr << "[ERROR] " << e.what() << '\n';
		return EXIT_FAILURE;
	}
	return EXIT_SUCCESS;
}

File: src/scripts/recallTable/checkDataset.py
from Dataset import Dataset
from pathlib import Path

N = "\n"

def main():
	try:
		dataDir = Path(__file__).parents[2] / "data"
		datasetPath = dataDir / "test.hdf5"
		descPath = dataDir / "testPy.txt"
		Dataset.fromHDF(datasetPath).writeDescription(descPath)
		print(f"Description of dataset {datasetPath} written to {descPath}.")

	except FileNotFoundError:
		print("HDF5 dataset not found. Run datasetGenerator.py first.")

if __name__ == "__main__":
	main()
